dm/syncer/error.go

```diff
@@ -211,13 +216,53 @@ func (s *Syncer) handleSpecialDDLError(tctx *tcontext.Context, err error, ddls [
 	}
 	// TODO: add support for downstream alter pk without schema
 
+	// it handles the operations for DDL when encountering `invalid connection` by waiting the asynchronous ddl to synchronize
+	waitAsyncDDL := func(tctx *tcontext.Context, err error, ddls []string, index int, conn *dbconn.DBConn, createTime int64) error {
+		if len(ddls) == 0 || index > len(ddls)-1 || errors.Cause(err) != mysql.ErrInvalidConn || createTime == -1 {
+			return err // return the original error
+		}
+
+		duration := 30
+		failpoint.Inject("ChangeDuration", func() {
+			duration = 1
+		})
+		ticker := time.NewTicker(time.Duration(duration) * time.Second)
+		defer ticker.Stop()
+
+		for {
+			status, err2 := getDDLStatusFromTiDB(tctx, conn, ddls[index], createTime)
```

Comment:
If downstream TiDB does not support this SQL syntax (maybe an old version of TiDB), we should not enter dead loop

Reply:
I changed it to if the retrieved status is not the other known possibilities, return the original error.